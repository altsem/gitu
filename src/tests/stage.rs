use super::*;

#[test]
fn staged_file() {
    let mut ctx = TestContext::setup_init();
    run(ctx.dir.path(), &["touch", "new-file"]);
    run(ctx.dir.path(), &["git", "add", "new-file"]);

    ctx.init_state();
    insta::assert_snapshot!(ctx.redact_buffer());
}

#[test]
fn stage_all_unstaged() {
    let ctx = TestContext::setup_init();
    commit(ctx.dir.path(), "firstfile", "testing\ntesttest\n");
    commit(ctx.dir.path(), "secondfile", "testing\ntesttest\n");

    fs::write(ctx.dir.child("firstfile"), "blahonga\n").unwrap();
    fs::write(ctx.dir.child("secondfile"), "blahonga\n").unwrap();
    snapshot!(ctx, "js");
}

#[test]
fn stage_all_untracked() {
    let ctx = TestContext::setup_init();
    run(ctx.dir.path(), &["touch", "file-a"]);
    run(ctx.dir.path(), &["touch", "file-b"]);
    snapshot!(ctx, "js");
}

#[test]
fn stage_removed_line() {
    let ctx = TestContext::setup_init();
    commit(ctx.dir.path(), "firstfile", "testing\ntesttest\n");
    fs::write(ctx.dir.child("firstfile"), "weehooo\nblrergh\n").unwrap();
    snapshot!(ctx, "jj<tab><ctrl+j><ctrl+j>s");
}

#[test]
fn stage_added_line() {
    let ctx = TestContext::setup_init();
    commit(ctx.dir.path(), "firstfile", "testing\ntesttest\n");
    fs::write(ctx.dir.child("firstfile"), "weehooo\nblrergh\n").unwrap();

    snapshot!(ctx, "jj<tab><ctrl+j><ctrl+j><ctrl+j><ctrl+j>s");
}

#[test]
fn stage_changes_crlf() {
    let ctx = TestContext::setup_init();
    commit(ctx.dir.path(), "testfile", "testing\r\ntesttest\r\n");
    fs::write(ctx.dir.child("testfile"), "test\r\ntesttest\r\n").expect("error writing to file");

    snapshot!(ctx, "jj<tab>");
}
