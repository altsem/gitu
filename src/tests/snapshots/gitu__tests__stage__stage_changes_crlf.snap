---
source: src/tests/stage.rs
expression: ctx.redact_buffer()
---
 On branch main                                                                 |
                                                                                |
 Unstaged changes (1)                                                           |
▌modified   testfile                                                            |
▌@@ -1,2 +1,2 @@                                                                |
▌-testing                                                                       |
▌+test                                                                          |
▌ testtest                                                                      |
                                                                                |
 Recent commits                                                                 |
 _______ main add testfile                                                      |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
styles_hash: df517b5688801d29
